src/counter.rs

```diff
@@ -78,6 +78,30 @@ impl Counter {
     }
 }
 
+pub struct CounterDec<'a> {
+    counter: &'a Counter,
+    amount: usize,
+}
+
+impl<'a> CounterDec<'a> {
+    pub fn new(counter: &'a Counter) -> Self {
```

Comment:
are there any concurrency issues accessing this counter?

Reply:
the counter is designed to be thread-safe (see the inc/dec impls). there is one instance shared by all tasks